lib/shoryuken/environment_loader.rb

```diff
@@ -43,13 +43,21 @@ def config_file_options
 
       fail ArgumentError, "The supplied config file #{path} does not exist" unless File.exist?(path)
 
-      if (result = YAML.load(ERB.new(IO.read(path)).result))
+      if (result = load_config_file(path))
         result.deep_symbolize_keys
       else
         {}
       end
     end
 
+    def load_config_file(path)
+      YAML.load(ERB.new(IO.read(path)).result)
+    rescue NameError => ex
+      fail(
+        "Rails and other constants not yet loaded. You can use ENV, or move your config to an initializer:\n#{ex.message}"
```

Comment:
Hi @nthj

That's very helpful. Thanks for this contribution.  I can see how this can be time-consuming to debug. 

I'm wondering if this can be misleading in case of other errors related to the ERB file, for example, bad syntax. How about explaining in the message that "Rails and other ..." can be one of the problems? 

How about also logging out the backtrace?

```ruby
logger.error { "Manager failed: #{ex.message}" }
logger.error { ex.backtrace.join("\n") } unless ex.backtrace.nil?
```

Reply:
Thanks @phstc, I'll make these changes and test them soon!